pkg/cmd/pulumi/new.go

```diff
@@ -246,47 +250,70 @@ func runNew(ctx context.Context, args newArgs) error {
 	fmt.Println()
 
 	// Load the project, update the name & description, remove the template section, and save it.
-	proj, root, err := readProject()
+	proj, path, err := readProjectWithPath()
+	root := filepath.Dir(path)
 	if err != nil {
 		return err
 	}
-	proj.Name = tokens.PackageName(args.name)
-	proj.Description = &args.description
-	proj.Template = nil
-	// Workaround for python, most of our templates don't specify a venv but we want to use one
-	if proj.Runtime.Name() == "python" {
-		// If the template does give virtualenv use it, else default to "venv"
-		if _, has := proj.Runtime.Options()["virtualenv"]; !has {
-			proj.Runtime.SetOption("virtualenv", "venv")
-		}
-	}
 
-	if err = workspace.SaveProject(proj); err != nil {
-		return fmt.Errorf("saving project: %w", err)
-	}
+	if filepath.Ext(path) == ".yaml" {
+		filedata, err := os.ReadFile(path)
+		if err != nil {
+			return err
+		}
+		var workspaceDocument yaml.Node
+		err = yaml.Unmarshal(filedata, &workspaceDocument)
+		if err != nil {
+			return err
+		}
 
-	if proj.Runtime.Name() == "yaml" {
-		projFile := filepath.Join(root, "Pulumi.yaml")
-		f, err := ioutil.ReadFile(projFile)
+		proj.Name = tokens.PackageName(args.name)
+		err = yamlutil.Insert(&workspaceDocument, "name", args.name)
 		if err != nil {
-			return fmt.Errorf("could not find Pulumi.yaml: %w", err)
+			return err
 		}
-		appendFileName := "Pulumi.yaml.append"
-		appendFile := filepath.Join(root, appendFileName)
-		m, err := ioutil.ReadFile(appendFile)
-		if err == nil {
-			f = append(f, m...)
-			err = ioutil.WriteFile(projFile, f, 0600)
-			if err != nil {
-				return fmt.Errorf("failed to write %s: %w", projFile, err)
-			}
-			err = os.Remove(appendFile)
-			if err != nil {
-				return fmt.Errorf("could not remove %s: %w", appendFileName, err)
+		proj.Description = &args.description
+		err = yamlutil.Insert(&workspaceDocument, "description", args.description)
+		if err != nil {
+			return err
+		}
+		proj.Template = nil
+		err = yamlutil.Delete(&workspaceDocument, "template")
+		if err != nil {
+			return err
+		}
+		if proj.Runtime.Name() == "python" {
+			// If the template does give virtualenv use it, else default to "venv"
+			if len(proj.Runtime.Options()) == 0 {
```

Comment:
dicey, virtualenv is the only option allowed on python at the moment... I think we should remove this hack well before we add a new option though